### PR TITLE
Android/update create path page

### LINKED ISSUE #364: Adding Tasks should be added where users can add milestones.
The prior design has only one addition button with one type of addible element as Milestone. The new design should be able to add both milestones and tasks. Two different button should be implemented as one button for each item type. The dynamic structure is somewhat advanced coding in flutter, so sufficient time and effort should be allocated for this property.

### LINKED ISSUE #365: Adding Tasks/Milestones segment should have proper naming and numbering protocols and arrange data accordingly.
As it is stated in the header, adding Tasks/Milestones segment should have proper naming and numbering protocols and arrange data accordingly. That means tasks and milestones should have incremented id's individually. As the order of those elements are important for the backend to follow hierarchy, correct ordering should be preserved.

### LINKED ISSUE #367: Topic addition part should have a search interface and users should be able to see topics with their description.
The prior design and implementation of Topic-Add section was not adequate. There were a lot of assumption and users were not able to select the topics that they wanted to add. After discussed in some meetings, the following conditions are decided to implement:
- Users should add topics one by one.
- Users should be able to search for a topic and see the result of their search. This result should include both topics and their description.
- Users should be able to choose Topic from search results.

### LINKED ISSUE #368: The Topic that are chosen should be shown clearly and should be removed if wanted.
There should be additional section under Topics segment which includes the topics that are already chosen for the learning path. It should contain all topics and users should have the chance to remove the topics if they are not happy. Dynamic implementation can be challenging.

### PR DESCRIPTION
**Pull request by Hazerank**
It is decided to change the general structure of the path in the Meeting 21.1 - 28.12.2021 . The implementation is started and followed in the Issue #363. In this pr request, the technical parts of the implementation is done. In order to focus on other features coding part, design issues are postponed.

To be specifically state, those following features are done, referencing regarding issues:

- Adding Task Feature:
  -  Adding Tasks items. (Issue #364) 
  -  Arranging Tasks/Milestones numbering. #365
- Updating Add-Topic Feature
  -  Topic addition  search interface. #367
  -  List of chosen topics. #368

### CODE REVIEW

**Review by fihlie — APPROVED** [review]
You could have implemented the tag features and milestone features in different branches. Besides that, with these changes create path page will be much more user friendly and more suitable for principles of the projects. Well done, keep up the good work.

File: source/android/Renaissance/lib/create_path.dart [hunk 1]
```diff
@@ -23,31 +26,99 @@ class _CreatePathPageState extends State<CreatePathPage> {
...
+      _typeOfController.removeAt(i);
     });
   }
 
+  void _searchTopic() async {
+    List<Tag> resultTags =
+        await HttpService.shared.searchTopic(topicController.text.trim());
+
+    topicController.clear();
+    return showDialog(
```
Comment (kursatbakis) [inline source/android/Renaissance/lib/create_path.dart:76]: not a good practice to return something in a void function.

File: source/android/Renaissance/lib/create_path.dart [hunk 2]
```diff
@@ -298,51 +481,38 @@ class _CreatePathPageState extends State<CreatePathPage> {
...
-                        milestones.add({
+                        items.add({
                           "title": _titleControllers[i].text,
-                          "body": _descControllers[i].text
+                          "body": _descControllers[i].text,
+                          "type": _typeOfController[i],
                         });
                       }
-                      print(milestones);
+                      print(items);
```
Comment (kursatbakis) [inline source/android/Renaissance/lib/create_path.dart:494]: delete the unnecessary prints

File: source/android/Renaissance/lib/create_path.dart [hunk 3]
```diff
@@ -298,51 +481,38 @@ class _CreatePathPageState extends State<CreatePathPage> {
...
-                      );
+                      // User response = await HttpService.shared.createPath(
+                      //   titleController.text,
+                      //   descriptionController.text,
+                      //   items,
+                      //   _image == null
+                      //       ? ""
+                      //       : FileConverter.getBase64StringFile(_image),
+                      //   sendTopic,
+                      // );
```
Comment (kursatbakis) [inline source/android/Renaissance/lib/create_path.dart:515]: remove commented out code

**Review by kursatbakis — CHANGES_REQUESTED** [review]
a few very small changes!

File: source/android/Renaissance/lib/create_path.dart (hunk 1 above)
Comment (Hazerank) [reply source/android/Renaissance/lib/create_path.dart:76]: Thanks, I should have missed that. Now the function has a proper return type.

File: source/android/Renaissance/lib/create_path.dart (hunk 2 above)
Comment (Hazerank) [reply source/android/Renaissance/lib/create_path.dart:494]: Okay, got rid of.

File: source/android/Renaissance/lib/create_path.dart (hunk 3 above)
Comment (Hazerank) [reply source/android/Renaissance/lib/create_path.dart:515]: This part is for backend connection and it is a framework for the future implementation. This lines will be implemented before the production code. So I will not remove them yet.

**Review by Hazerank — COMMENTED** [review]
Changes are done as Kürşat offered.

**Review by kursatbakis — APPROVED** [review]
thanks for applying the changes. What you did in this PR is a really heavy task. Congratulations

**Pull request merged**